File: src/main/java/ru/romanow/serialization/model/InnerData.java
package ru.romanow.serialization.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.xml.bind.annotation.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@XmlAccessorType(XmlAccessType.FIELD)
public class InnerData {

    @XmlValue
    private String code;

    @XmlAttribute(name = "priority")
    private Integer priority;
}

File: src/main/java/ru/romanow/serialization/model/InnerData.kt
package ru.romanow.serialization.model

import jakarta.xml.bind.annotation.XmlAccessType
import jakarta.xml.bind.annotation.XmlAccessorType
import jakarta.xml.bind.annotation.XmlAttribute
import jakarta.xml.bind.annotation.XmlValue

@XmlAccessorType(XmlAccessType.FIELD)
data class InnerData(
    @XmlValue
    var code: String? = null,

    @XmlAttribute(name = "priority")
    var priority: Int? = null
)
